### PR TITLE
Perception without nodelets

### PR DESCRIPTION
**Pull request by flamma7**
Ability to run perception as or not as a nodelet. When code fails as a nodelet, there is usually very little insight into the source of the problem, often not even which node crashed. Running nodes normally gives us the typical debug output from a program when it crashes.

### CODE REVIEW

File: cusub_perception/detection_tree/src/detection_tree_node.cpp [hunk 1]
```diff
@@ -6,10 +6,23 @@ using namespace cv;
...
+        cuprint("\033[95mNOT\033[0m running as nodelet");
+        init(nh);
+    }
+}
+
+void DetectionTree::init(ros::NodeHandle& nh)
+{
     dvector_num = 0;
-    
+    sub_name = "leviathan"; // TODO Pull from config
```
Comment (DrYerzinia) [inline cusub_perception/detection_tree/src/detection_tree_node.cpp]: Bad luke, bad..

Comment (DrYerzinia) [reply cusub_perception/detection_tree/src/detection_tree_node.cpp]: If you are feeling that lazy just ask ROS for the current namespace

File: cusub_perception/localizer/src/localizer.cpp [hunk 2]
```diff
@@ -106,6 +130,19 @@ namespace localizer_ns
...
+    }
 }
 
 PLUGINLIB_EXPORT_CLASS(localizer_ns::Localizer, nodelet::Nodelet);
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "localizer_node");
+  localizer_ns::Localizer l;
+  l.set_not_nodelet();
```
Comment (DrYerzinia) [inline cusub_perception/localizer/src/localizer.cpp:145]: Surely there is less stupid way to check this?

**Review by DrYerzinia — CHANGES_REQUESTED** [review]
Wrong approach, redo by rewriting all code to run in ROS 2...

lol, no jk, but seriously, no direct references to what sub this is running in.

File: cusub_perception/detection_tree/src/detection_tree_node.cpp (hunk 1 above)
Comment (flamma7) [reply cusub_perception/detection_tree/src/detection_tree_node.cpp]: yeah lol will be fixed when we have a second sub

Comment (flamma7) [reply cusub_perception/detection_tree/src/detection_tree_node.cpp]: damn changes requested... I guess I'll fix it

File: cusub_perception/localizer/src/localizer.cpp (hunk 2 above)
Comment (flamma7) [reply cusub_perception/localizer/src/localizer.cpp:145]: maybe, it's not great but it does get the job done

**Comment by flamma7** [comment]
yeah let me just migrate the entire code base to ROS2 in this next commit

**Comment by flamma7** [comment]
K I'll spend the spend the 30s to fix that

**Review by DrYerzinia — APPROVED** [review]
Now I can sleep soundly at night knowing that if for some reason I run perception on the microSub I wont have to deal with this stupid problem.

**Review by skhadem — APPROVED** [review]
Just curious, what's the reason to not run as nodelet? Debugging reasons?

**Comment by flamma7** [comment]
Yes, when nodelets crash they provide no insight into what the problem was. Also we can't see the connections between nodes inside the nodelets through rqt graph

**Review by APChehadi — APPROVED** [review]
Very epic

**Pull request merged**